src/arduino.cc/builder/phases/sizer.go

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of Arduino Builder.
+ *
+ * Arduino Builder is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As a special exception, you may use this file as part of a free software
+ * library without restriction.  Specifically, if other files instantiate
+ * templates or use macros or inline functions from this file, or you compile
+ * this file and link it with other files to produce an executable, this
+ * file does not by itself cause the resulting executable to be covered by
+ * the GNU General Public License.  This exception does not however
+ * invalidate any other reasons why the executable file might be covered by
+ * the GNU General Public License.
+ *
+ * Copyright 2016 Arduino LLC (http://www.arduino.cc/)
+ */
+
+package phases
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strconv"
+
+	"arduino.cc/builder/builder_utils"
+	"arduino.cc/builder/constants"
+	"arduino.cc/builder/i18n"
+	"arduino.cc/builder/types"
+	"arduino.cc/properties"
+)
+
+type Sizer struct{}
+
+func (s *Sizer) Run(ctx *types.Context) error {
+	buildProperties := ctx.BuildProperties
+	verbose := ctx.Verbose
+	warningsLevel := ctx.WarningsLevel
+	logger := ctx.GetLogger()
+
+	err := checkSize(buildProperties, verbose, warningsLevel, logger)
+	if err != nil {
+		return i18n.WrapError(err)
+	}
+
+	return nil
+}
+
+func checkSize(buildProperties properties.Map, verbose bool, warningsLevel string, logger i18n.Logger) error {
+
+	properties := buildProperties.Clone()
+	properties[constants.BUILD_PROPERTIES_COMPILER_WARNING_FLAGS] = properties[constants.BUILD_PROPERTIES_COMPILER_WARNING_FLAGS+"."+warningsLevel]
+
+	maxTextSizeString := properties["upload.maximum_size"]
+	maxDataSizeString := properties["upload.maximum_data_size"]
+
+	if maxTextSizeString == "" {
+		return nil
+	}
+
+	logLevel := constants.LOG_LEVEL_INFO
+
+	maxTextSize, _ := strconv.Atoi(maxTextSizeString)
+	maxDataSize := -1
+
+	if maxDataSizeString != "" {
+		maxDataSize, _ = strconv.Atoi(maxDataSizeString)
```

Comment:
What does Atoi return on an error? Is the first return value even defined in that case? Or should the err value (that is now ignored) be checked instead? This also applies to some other Atoi calls.